app/src/shared/stores/streamAbilities.ts

```diff
@@ -80,49 +82,55 @@ const useStreamAbilitiesStore = create<Store>((set, get) => {
                 return
             }
 
+            let result = false
+
             try {
                 toggleFetching(streamId, account, permission, true)
 
-                let stream: Stream | undefined
+                let stream: Stream | null | undefined
 
                 try {
-                    stream = await streamrClient.getStream(streamId)
+                    stream = await streamrClient.getStream(
+                        streamId,
+                    )
                 } catch (e) {
-                    // Do nothing here.
-                }
+                    if (isMessagedObject(e) && /not_found/i.test(e.message)) {
+                        throw new StreamNotFoundError(streamId)
+                    }
 
-                if (!stream) {
-                    throw new Error('Stream not found')
+                    throw e
                 }
 
-                const value = await stream.hasPermission(account === address0 ? {
-                    permission,
-                    public: true,
-                } : {
-                    user: account,
-                    permission,
-                    allowPublic: true,
-                })
-
+                result = await stream.hasPermission(
+                    account === address0
+                        ? {
+                              permission,
+                              public: true,
+                          }
+                        : {
+                              user: account,
+                              permission,
+                              allowPublic: true,
+                          },
+                )
+            } finally {
```

Comment:
See https://github.com/streamr-dev/core-frontend/pull/1490/commits/38cee3c4facdaecfbd55c85b6d045c2c254ad2a1 for more details.